### PR TITLE
[usb] Update type for transfer errors

### PR DESCRIPTION
**Pull request by dsanders11**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/tessel/node-usb/blob/3db45bb9d8afca69e9337e3134c764ef80477656/src/node_usb.cc#L298
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/usb/index.d.ts [hunk 1]
```diff
@@ -8,6 +8,10 @@
 
 import { EventEmitter } from "events";
 
+export class libusbException extends Error {
```
Comment (thegecko) [inline types/usb/index.d.ts]: Please follow the capitalisation naming convention for classes used elsewhere in this file:

`LibUSBException`

**Review by thegecko — CHANGES_REQUESTED** [review]
Apart from the name and test updates, looks good.

**Comment by thegecko** [comment]
@dsanders11 The tests fail because you need to use your new type (`usb.libusbException`) in place of the `string` type for the error variables on these lines:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/usb/usb-tests.ts#L15

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/usb/usb-tests.ts#L106

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/usb/usb-tests.ts#L120

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/usb/usb-tests.ts#L121

**Comment by dsanders11** [comment]
@thegecko, thanks for the line number. Addressed test errors and the name casing.

**Review by thegecko — APPROVED** [review]
Awesome, LGTM

**Pull request merged**